lib/open_trip_planner_client.ex

```diff
@@ -12,41 +12,19 @@ defmodule OpenTripPlannerClient do
   """
   @behaviour OpenTripPlannerClient.Behaviour
 
-  alias OpenTripPlannerClient.{ItineraryTag, ParamsBuilder, Parser, PlanParams, Util}
+  alias OpenTripPlannerClient.{ItineraryTag, Parser, PlanParams, Util}
 
   require Logger
 
   @plan_query File.read!("priv/plan.graphql")
 
   @impl OpenTripPlannerClient.Behaviour
   @doc """
-  Generate a trip plan with the given endpoints and options.
+  Generate a trip plan with the given endpoints and options. Supports customizing which tags are applied to the results.
   """
-  def plan(from, to, opts) do
-    {tags, opts} = Keyword.pop(opts, :tags, default_tags(opts))
-
-    case ParamsBuilder.build_params(from, to, opts) do
-      {:ok, params} ->
-        params
-        |> PlanParams.new()
-        |> do_plan(tags)
-
-      error ->
-        error
-        |> inspect()
-        |> Logger.error(%{from: from, to: to, opts: opts})
-
-        error
-    end
-  end
-
-  @impl OpenTripPlannerClient.Behaviour
   def plan(params, tags \\ nil) do
     tags = if tags, do: tags, else: default_tags(params)
-    do_plan(params, tags)
-  end
 
-  defp do_plan(params, tags) do
```

Comment:
Love this! this whole `foo/do_foo` pattern that I keep seeing everywhere is so confusing!